File: Day - 1/Quartiles/try.java
import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Arrays;
import java.util.Scanner;

public class Quartiles {

    static double getMedian(int[] arr) {
        Arrays.sort(arr);
        int mid = arr.length / 2;
        if (arr.length % 2 != 0) {
            return arr[mid];
        } else {
            return (arr[mid] + arr[mid - 1]) / 2.0;
        }
    }

    static int[] quartiles(int[] arr) {
        Arrays.sort(arr);
        int n = arr.length;
        int mid = n / 2;

        int q1, q2, q3;
        if (n % 2 != 0) {
            q1 = (int) getMedian(Arrays.copyOfRange(arr, 0, mid));
            q2 = (int) getMedian(arr);
            q3 = (int) getMedian(Arrays.copyOfRange(arr, mid + 1, n));
        } else {
            q1 = (int) getMedian(Arrays.copyOfRange(arr, 0, mid));
            q2 = (int) getMedian(arr);
            q3 = (int) getMedian(Arrays.copyOfRange(arr, mid, n));
        }

        return new int[]{q1, q2, q3};
    }

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] data = new int[n];

        for (int i = 0; i < n; i++) {
            data[i] = scanner.nextInt();
        }

        int[] res = quartiles(data);

        for (int i = 0; i < res.length; i++) {
            bufferedWriter.write(String.valueOf(res[i]));

            if (i != res.length - 1) {
                bufferedWriter.newLine();
            }
        }

        bufferedWriter.newLine();
        bufferedWriter.close();

        scanner.close();
    }
}

File: Day - 1/Quartiles/try.py
#!/bin/python3

import math
import os
import random
import re
import sys

def getMedian(arr):
    arr.sort()
    mid = len(arr) // 2
    if len(arr) % 2 != 0:
        return arr[mid]
    else:
        return (arr[mid] + arr[mid-1]) / 2

def quartiles(arr):
    arr.sort()
    n = len(arr)
    mid = n // 2
    
    if n % 2 != 0:
        q1 = getMedian(arr[:mid])
        q2 = getMedian(arr)
        q3 = getMedian(arr[mid+1:])
    else:
        q1 = getMedian(arr[:mid])
        q2 = getMedian(arr)
        q3 = getMedian(arr[mid:])
        
    return [int(x) for x in [q1, q2, q3]]

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    data = list(map(int, input().rstrip().split()))

    res = quartiles(data)

    fptr.write('\n'.join(map(str, res)))
    fptr.write('\n')

    fptr.close()
